### ISSUE TITLE
Seeking for metrics function for pseudoknot 

### ISSUE THREAD

**Issue by ComDec** [gongtiansu/KnotFold#2]
Hi there, can you provide your code for evaluate the performance of pseudo knot prediction?

**Comment by gongtiansu** [comment]
Hello, we use the function "precision_recall_fscore_support" from the sklearn.metrics module in Python to evaluate the binary classification performance for predicting RNA secondary structure. Here is the code, where truth, pred are upper triangular binary matrices of size L * L :


`    
        

    for status in ["Overall", "Pseudoknot-free", "Crossing-pseudoknot", "Pseudoknotted" ]:     

        prec, recall, f1, _ = precision_recall_fscore_support(    

            y_true=truth[pseudoknot == status],    

            y_pred=pred[pseudoknot == status],    

            zero_division=0,    

             average="binary",    

        )    

`

**Comment by gongtiansu** [comment]
> Hi there, can you provide your code for evaluate the performance of pseudo knot prediction?

Hello, we treat RNA secondary-structure prediction as a binary classification problem and we use the function "precision_recall_fscore_support" from the sklearn.metrics module in Python to assess the performance of predicting RNA secondary structures. 

Here is the code, where pred and truth are upper triangular binary matrix of size L * L, where L is the length of the sequence:

    prec, recall, f1, _ = precision_recall_fscore_support(
        y_true=truth, y_pred=pred, zero_division=0, average="binary"
     )

We also evaluate the base-pair level performance, by categorizing each base pair into three distinct labels ("Pseudoknot-free", "Crossing-pseudoknot", and "Pseudoknotted"), and calculating precision, recall, and F1 scores for each label set.

Here is the code:


    for status in ["Pesudoknot-free", "Crossing-pseudoknot", "Pseudoknotted"]:    
        prec, recall, f1, _ = precision_recall_fscore_support(    
            y_true=truth[pseudoknot == status],    
            y_pred=pred[pseudoknot == status],    
            zero_division=0,    
            average="binary",    
        )

**Comment by skoc** [comment]
Hi, thank you, this post helped to understand your evaluation, if possible can you share your evaluation script?
